llama-index-core/llama_index/core/base/agent/types.py

```diff
@@ -172,8 +172,8 @@ class Config:
         ),
     )
 
-    callback_manager: CallbackManager = Field(
-        default_factory=CallbackManager,
+    callback_manager: Optional[CallbackManager] = Field(
+        default_factory=lambda: CallbackManager([]),
```

Comment:
I feel like I originally had a lambda here and ran into pickling issues. Can we use an actual function instead of a lambda? (surprised that the previous version broke, sadge)

Reply:
ah, actually maybe not, I see lambda used in other places already

Reply:
i think i've had problems with using lambda when I just return the cls and forget to actually construct an object lol

Reply:
Default factory treats it as a callable 🤔 but yea, idk lol

Reply:
Oh yeah. I think I was doing something like:

lambda: MyClass

Which the lambda is a callable but it returns a type (ie class) instead of an object?